include/qpid/dispatch/iterator.h

```diff
@@ -180,6 +180,8 @@ void qd_address_iterator_set_phase(qd_field_iterator_t *iter, char phase);
  */
 void qd_address_iterator_override_prefix(qd_field_iterator_t *iter, char prefix);
 
+void qd_address_iterator_set_length(qd_field_iterator_t *iter, int length);
```

Comment:
You should add a comment to document this new method.
What happens if this method is called with a length that is greater than the current iterator's length?